src/ui/server/controllers/insurance/policy-and-export/multiple-contract-policy/index.ts

```diff
@@ -104,3 +107,68 @@ export const get = async (req: Request, res: Response) => {
     return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
   }
 };
+
+/**
+ * post
+ * Check Multiple contract policy validation errors and if successful, redirect to the next part of the flow.
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} Next part of the flow or error page
+ */
+export const post = async (req: Request, res: Response) => {
+  const { application } = res.locals;
+
+  if (!application) {
+    return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+  }
+
+  const { referenceNumber } = req.params;
+  const refNumber = Number(referenceNumber);
+
+  const validationErrors = generateValidationErrors(req.body);
+
+  if (validationErrors) {
+    try {
+      const currencies = await api.external.getCurrencies();
+
+      if (!currencies || !currencies.length) {
+        return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+      }
+
+      const mappedCurrencies = mapCurrencies(currencies);
+
+      return res.render(TEMPLATE, {
+        ...insuranceCorePageVariables({
+          PAGE_CONTENT_STRINGS: PAGES.INSURANCE.POLICY_AND_EXPORTS.MULTIPLE_CONTRACT_POLICY,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...pageVariables(refNumber),
+        application: mapApplicationToFormFields(application),
+        submittedValues: req.body,
```

Comment:
in the future, I think we should santise the req.body before passing it back? future PR

Reply:
Interesting, good idea - i'm wondering what value we'd get from that though - because when rendered, it would be transformed back into what it was before it was posted? We'd also need a different (or adapt existing) "sanitise data" function because in this scenario, we wouldn't want to for example, remove commas from a number